src/database.c

```diff
@@ -203,9 +231,17 @@ static int write_db(const char *index, const char *data)
 		return 2;
 	}
 
-	key.mv_data = (void *)index;
-	key.mv_size = strlen(index);
-	value.mv_data = (void *)data;
+    len = strlen(index);
+    if (len > MAX_PATH_LEN) { /* key size is greater than LMDB key limit */
+        hash = path_to_hash(index);
```

Comment:
There's no error path here if path_to_hash() does happen to return NULL. Do we care?